OpenEdXMobile/src/main/java/org/edx/mobile/view/dialog/CourseModalDialogFragment.kt

```diff
@@ -210,17 +206,16 @@ class CourseModalDialogFragment : DialogFragment() {
 
     private fun onProductPurchased(purchaseToken: String) {
         lifecycleScope.launch {
-            executeOrder(purchaseToken)
+            iapViewModel.setPurchaseToken(purchaseToken)
+            iapViewModel.showFullScreenLoader()
+            dismiss()
         }
     }
 
-    private fun executeOrder(purchaseToken: String) {
-        iapViewModel.executeOrder(purchaseToken = purchaseToken)
-    }
-
     private fun showUpgradeErrorDialog(
         @StringRes errorResId: Int = R.string.general_error_message
     ) {
+        if (!isAdded) return;
         AlertDialogFragment.newInstance(
             getString(R.string.title_upgrade_error),
             getString(errorResId),
```

Comment:
Just a confirmation -> are we doing nothing when the user taps on the close button of ErrorDialog? i.e, dismiss the ValueProp or something like this.

Reply:
Yeap, we will do **nothing** on close if the error occurs on `basket`, `checkout`, or `SDK` errors. So the user can try again if he wants to. 
ps: We will close the modal only when we are `unable to fetch the course price` or if the error occurs on the `execute` or `refresh` call.